lib/LongVectorLoweringPass.cpp

```diff
@@ -426,7 +634,8 @@ bool LongVectorLoweringPass::runOnFunction(Function &F) {
 
   bool Modified = (FunctionToVisit != &F);
   for (Instruction &I : instructions(FunctionToVisit)) {
-    Modified |= (visit(&I) != nullptr);
+    // Use the Value overload of visit to ensure cache is used.
+    Modified |= (visit(static_cast<Value *>(&I)) != nullptr);
```

Comment:
Does LLVM's `cast` not suffice here? If it does it is preferable to `static_cast`.